pytest_reportportal/plugin.py

```diff
@@ -14,24 +16,48 @@
     PYTEST_HAS_LOGGING_PLUGIN = False
 
 
+def is_master(config):
+    """
+    True if the code running the given pytest.config object is running in a xdist master
+    node or not running xdist at all.
+    """
+    return not hasattr(config, 'slaveinput')
+
+
+@pytest.mark.optionalhook
+def pytest_configure_node(node):
+    node.slaveinput['py_test_service'] = pickle.dumps(node.config.py_test_service)
+
+
 def pytest_sessionstart(session):
     if session.config.getoption('--collect-only', default=False) is True:
         return
 
-    PyTestService.init_service(
-        project=session.config.getini('rp_project'),
-        endpoint=session.config.getini('rp_endpoint'),
-        uuid=session.config.getini('rp_uuid'),
-        log_batch_size=int(session.config.getini('rp_log_batch_size')),
-        ignore_errors=bool(session.config.getini('rp_ignore_errors')),
-        ignored_tags=session.config.getini('rp_ignore_tags'),
-    )
+    if is_master(session.config):
+        session.config.py_test_service.init_service(
+            project=session.config.getini('rp_project'),
+            endpoint=session.config.getini('rp_endpoint'),
+            uuid=session.config.getini('rp_uuid'),
+            log_batch_size=int(session.config.getini('rp_log_batch_size')),
+            ignore_errors=bool(session.config.getini('rp_ignore_errors')),
+            ignored_tags=session.config.getini('rp_ignore_tags'),
+        )
+
+        session.config.py_test_service.start_launch(
+            session.config.option.rp_launch,
+            tags=session.config.getini('rp_launch_tags'),
+            description=session.config.getini('rp_launch_description'),
+        )
+        if session.config.pluginmanager.hasplugin('xdist'):
+            wait_launch(session.config.py_test_service.RP.rp_client)
 
-    PyTestService.start_launch(
-        session.config.option.rp_launch,
-        tags=session.config.getini('rp_launch_tags'),
-        description=session.config.option.rp_launch_description,
-    )
+
+def wait_launch(rp_client):
+    timeout = time.time() + 10
```

Comment:
I would avoid "magic" numbers and create file/module level const for `10`.

Reply:
Yes you are right. It's done.
+ I changed the example of working with RPLogHandler in README